### PR TITLE
module updating refactored

### PR DESCRIPTION
**Pull request by wawa1474**
updating modules will now go through all inventory slots regardless of inventory size

### CODE REVIEW

**Comment by wawa1474** [comment]
this will make it so all modules will be updated in any machine or chest with any size inventory

**Comment by wawa1474** [comment]
I'm working on windows and notepad++ and the indents and stuff looks good on my end but looks messed up in githubs view, and I don't know how to fix that. Sorry...

File: alien-module/control.lua [hunk 1]
```diff
@@ -73,45 +73,52 @@ function update_gui()
...
+        elseif entityType == "machine" then
+            inventory = entity.get_module_inventory() --grab a machine's inventory
+        elseif entityType == "player" then
+            inventory = entity.get_inventory(defines.inventory.player_main) --grab a player's inventory
+        else
+            return --error entity type not defined
+        end
 
-        for i = 1, 10, 1 do
+        for i = 1, #inventory, 1 do --loop through all of the entity's inventory slots
```
Comment (renoth) [inline alien-module/control.lua]: what is #inventory?

Comment (renoth) [reply alien-module/control.lua]: oh i see its the length of the array

Comment (wawa1474) [reply alien-module/control.lua]: it's the number (#) of slots in the variable [inventory]

**Comment by renoth** [comment]
Thanks a lot! I will playtest and create a new version soon.

File: alien-module/control.lua (hunk 1 above)
Comment (wawa1474) [reply alien-module/control.lua]: https://lua-api.factorio.com/latest/LuaInventory.html

it shows it here

operator # :: uint [R] - Get the number of slots in this inventory.

Comment (wawa1474) [reply alien-module/control.lua]: where do you get the lua formatter?

Comment (renoth) [reply alien-module/control.lua]: i use the intellij standard lua formatter

**Pull request merged**